src/Components/GenerateQuestionsForm/index.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  Box,
+  FormControl,
+  Grid,
+  InputLabel,
+  MenuItem,
+  Select,
+  SelectChangeEvent,
+  TextField,
+  Button,
+} from "@mui/material";
+import { useAtom } from "jotai";
+import React, { useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { componentInDisplayAtom } from "../../Views/GenerateQuestions/atoms";
+import { LoadingButton } from "@mui/lab";
+
+const GenerateQuestionsForm = () => {
+  const [componentInDisplay, setComponentInDisplay] = useAtom(
+    componentInDisplayAtom
+  );
+  const { quizId } = useParams();
+  const navigate = useNavigate();
+
+  const [generateSettings, setGenerateSettings] = useState({
+    quizId: quizId,
+    amount: 5,
+    question_type: "multiple-choice",
+    instructions: "",
+  });
+
+  const handleInputChange = (
```

Comment:
It will probably be more elegant to separate these two functions. There's no reason for them to be together.